src/aleph/http/multipart.clj

```diff
@@ -146,3 +158,94 @@
           (.addBodyHttpData encoder attr))))
     (let [req' (.finalizeRequest encoder)]
       [req' (when (.isChunked encoder) encoder)])))
+
+(defmulti http-data->map
+  (fn [^InterfaceHttpData data]
+    (.getHttpDataType data)))
+
+(defmethod http-data->map InterfaceHttpData$HttpDataType/Attribute
+  [^Attribute attr]
+  (let [content (.getValue attr)]
+    {:part-name (.getName attr)
+     :content content
+     :name nil
+     :charset (-> attr .getCharset .toString)
+     :mime-type nil
+     :transfer-encoding nil
+     :memory? (.isInMemory attr)
+     :file? false
+     :file nil
+     :size (count content)}))
+
+(defmethod http-data->map InterfaceHttpData$HttpDataType/FileUpload
+  [^FileUpload data]
+  (let [memory? (.isInMemory data)]
+    {:part-name (.getName data)
+     :content (when memory?
```

Comment:
If it's in a file, why not just give a memory-mapped byte-buffer?

Reply:
Do you mean to use `ByteBuffer` in both cases?  We can do `InputStream` for both: right now content is converted to `InputStream` for in-memory uploads, but we can easily create an input stream directly from a temp file. I'm not sure what the most convenient way.

Reply:
Mostly I was confused why we were forcing the consumer of the API to check whether the data was in a file or not, but now that I think about it a bit more, I'm guessing it's because we don't want to allocate a file descriptor without them asking for it.  Is that right?